### DISCUSSION TITLE
Is there any way to use custom DNS servers from cloudshell

### DISCUSSION THREAD

**Discussion by brwilkinson** [Azure/CloudShell#128] [Q&A]
I have Cloudshell enabled in Virtual Network, via relay with privatelink and storage account with Service Endpoints.

I have custom DNS servers set of my VNET, however the CloudShell will just use Azure DNS.

Because of this I am unable to connect to my Azure Resources from cloudshell on the VNET.

- The root cause of why it doesn't work, is because I have my private DNS zones setup in a HUB VNET, which host some DNS forwarder servers, then I set other VNET DNS to use those forwarders, they also have the DNS Private Zones linked only in that HUB VNET.

**Comment by theJasonHelmick** [comment]
Thank you Ben for this request.  We are investigating this possibility and will answer back as we find a solution forward. My anticipation is that we will be able to support this in the future.

**Reply by audunsolemdal** [reply]
@theJasonHelmick any status update on this?

I have the same case where I have VNET-integrated cloud shell and want to be able to reach my kubernetes clusters and container registry which are behind a private endpoint. My VNET spoke is connected to a hub with azure firewall which functions as the DNS server.

Am I correct in my assumption that the VNET integration will not work with anything that requires overriding the default Azure DNS servers? If that is the case there should be a disclaimer regarding limitations somewhere in this doc IMO
https://learn.microsoft.com/en-us/azure/cloud-shell/vnet/overview

**Comment by priyaananthasankar** [comment]
@brwilkinson : Cloud shell does not enforce DNS to go via Azure DNS, if you have a custom DNS server, then ensure your private DNS zone is aware of them (or) you could use Azure private DNS resolver. Can you share a high level diagram of this infrastructure?
